packages/kit/src/runtime/client/client.js

```diff
@@ -403,10 +403,10 @@ export function create_client({ target, base, trailing_slash }) {
 		let data = {};
 		let data_changed = false;
 		for (let i = 0; i < filtered.length; i += 1) {
-			Object.assign(data, filtered[i].data);
+			data = { ...data, ...filtered[i].data };
 			// Only set props if the node actually updated. This prevents needless rerenders.
 			if (!current.branch.some((node) => node === filtered[i])) {
-				result.props[`data_${i}`] = filtered[i].data;
+				result.props[`data_${i}`] = data;
```

Comment:
Is this logic still correct? If any parent updates, so does the data so we likely can no longer rely on "this node didn't change", but rather "everything up to and including this node didn't change"

Reply:
oops, yep — nice save

Reply:
https://github.com/sveltejs/kit/pull/6050/commits/f0823cb459f22d4c22549bb490680f0c6a165ade